### DISCUSSION TITLE
important requirement

### DISCUSSION THREAD

**Discussion by technoelias** [tyron12233/CodeAssist#502] [Ideas]
Hello Tyron. First of all I want to tell you that I am very satisfied with CodeAssist, and I appreciate all the effort you are putting into continuing to develop this wonderful IDE.

Since I unfortunately do not have a desktop device among my belongings, I am using CodeAssist to develop my projects. But I have a special use case, which requires you to consider what I'm going to propose next, for when you're ready to release the next version of CodeAssist.

As you know, Google Play app publishing policies force you not to allow CodeAssist users to disable storage restrictions that were added starting with Android 11.<br/>
So I came up with an easy working solution that you can take to achieve the desired goal without getting into trouble with Google Play, since that doesn't require you to request the `android.permission.MANAGE_EXTERNAL_STORAGE` permission. Please read the following lines carefully.
1. Change the value of `targetSdkVersion` property to 29
2. Build an APK file, and upload it to Github.
3. Change the value of `targetSdkVersion` property to the value it had before you did the above.
4. Build an AAB file, and upload it to Google Play.

#### In case you consider doing the above, you should also do the following.

Implement the following logic when the app starts: `if(Build.VERSION.SDK_INT>=30 && Environment.isExternalStorageLegacy()){/* not installed from Goggle Play, can enable the file browser dialog */}`

Explain something like the following to the user: **If you want to be able to access any directory of your device's internal storage, you would need to download an alternate version of CodeAssist from Github. And if it does, you'll need to uninstall this version first, so you won't have any trouble installing that alternate version.**

**Comment by PranavPurwar** [comment]
people that download with play store might not have knowledge about github or they might not be log in-nd on github so they wouldn't be abloe to download the app. and in addition to that, github won't tell them to login manually

**Reply by technoelias** [reply]
Your claim is incorrect.  Since the people who downloaded CodeAssist are programmers, it's quite unlikely that they don't know a bit about Github, so if you use your head more, you'll see that there's no reason to reject my proposal.

**Reply by PranavPurwar** [reply]
I didn't make a 'claim' :skull: And if you've been here since the beginning of CodeAssist, you'll know that many sketchware [pro] users have migrated to CodeAssist. Like, do you really think a beginner like who just wanted to begin creating android apps, would be called "programmers"? And tbh, knowing Github and being log-inned to GitHub are very different. And well, be respectful. You're not like a super-professional guy to be guiding others.

**Reply by technoelias** [reply]
Hello Pranav.  I'm sorry for disrespecting you.
 What I meant to tell you with my previous message was that: If a person doesn't know how to use Github, he will download the restricted version of CodeAssist from the Play Store.  But let those of us who do know how to use github download an unrestricted version of CodeAssist.

**Comment by tyron12233** [comment]
Thanks for the idea, there will be a separate flavor for the github version and the google play store. The former will have a lower target sdk version and android version checks removed. It will be implemented right after I finish working on gradle. In the meantime, you can modify the source code yourself and remove the checks.

**Reply by technoelias** [reply]
Thanks a lot.
